containerd-http-proxy/configure_http_proxy.yaml

```diff
@@ -0,0 +1,88 @@
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: containerd-http-proxy
+spec:
+  selector:
+    matchLabels:
+      name: containerd-http-proxy
+  template:
+    metadata:
+      labels:
+        name: containerd-http-proxy
+    spec:
+      hostPID: true
+      containers:
+      - name: startup-script
+        image: gke.gcr.io/debian-base:bookworm-v1.0.0-gke.1
+        imagePullPolicy: IfNotPresent
+        securityContext:
+          privileged: true
+        command:
+        - /bin/sh
+        - -c
+        - |
+          set -e
+          set -u
+
+          validate_proxy() {
+            input_string=$1
+
+            if echo "$input_string" | grep -q ' '; then
+                echo "Error: Input cannot contain spaces. Input: '$input_string'"
+                exit 1
+            fi
+
+            if echo "$input_string" | sed 1d | grep -q .; then
+                echo "Error: Input cannot contain newline. Input: '$input_string'"
+                exit 1
+            fi
+
+            if echo "$input_string" | grep -q -e '"' -e "'"; then
+              echo "Error: Input cannot contain quotes. Input: '$input_string'"
+              exit 1
+            fi
+          }
+
+          validate_proxy "${HTTP_PROXY:-}"
+          validate_proxy "${HTTPS_PROXY:-}"
+          validate_proxy "${NO_PROXY:-localhost}"
+
+          STARTUP_SCRIPT=$(cat <<'END'
```

Comment:
Rather than the multiple layers of indirection here, it seems like it would be a lot simpler to just mount `/etc/systemd/system/containerd.service.d` as a host volume and write the script directly from here. Is there a reason you can't use a host mount?

Reply:
That's good idea. Updated.